### PR TITLE
feat: add NewJsonResponseOrPanic function to simplify calls with ResponderFromMultipleResponses

### PR DESCRIPTION
**Pull request by kilianpaquier**
Simplify calls to `ResponderFromMultipleResponses` for JSON body responses with `NewJsonResponseOrPanic`.

Just as the same as `NewJsonResponderOrPanic` and `NewJsonResponder`, it's an addition to simplify some calls and avoid local variables with error handling.

### CODE REVIEW

File: response.go [hunk 1]
```diff
@@ -586,6 +586,31 @@ func NewJsonResponse(status int, body any) (*http.Response, error) { // nolint:
 	return response, nil
 }
 
+// NewJsonResponseOrPanic is like [NewJsonResponse] but panics in case or error.
```
Comment (maxatome) [inline response.go]:
```suggestion
// NewJsonResponseOrPanic is like [NewJsonResponse] but panics in case of error.
```

File: response.go [hunk 2]
```diff
@@ -586,6 +586,31 @@ func NewJsonResponse(status int, body any) (*http.Response, error) { // nolint:
...
+//
+// It simplifies the call of [ResponderFromMultipleResponses], avoiding the use of a
```
Comment (maxatome) [inline response.go]:
```suggestion
// It simplifies the call of [ResponderFromMultipleResponses], avoiding the
// use of a temporary variable and an error check, and so can be used in such
```

File: response.go [hunk 3]
```diff
@@ -586,6 +586,31 @@ func NewJsonResponse(status int, body any) (*http.Response, error) { // nolint:
...
+// temporary variable and an error check, and so can be used in such context:
+//
+//	httpmock.RegisterResponder(
+//	  "GET",
+//	  "/test/path",
+//	  httpmock.ResponderFromMultipleResponses([]*http.Response{
+//		httpmock.NewJsonResponseOrPanic(200, &MyFirstResponseBody),
```
Comment (maxatome) [inline response.go]: Only the first column should be a TAB, following columns should be 2 spaces per indent, like you did the line just above:
```suggestion
//	    httpmock.NewJsonResponseOrPanic(200, &MyFirstResponseBody),
```

File: response.go [hunk 4]
```diff
@@ -586,6 +586,31 @@ func NewJsonResponse(status int, body any) (*http.Response, error) { // nolint:
...
+//		httpmock.NewJsonResponseOrPanic(200, &MySecondResponseBody),
+//	  }),
+//	)
+//
+// To pass the content of an existing file as body use [File] as in:
+//
+//	httpmock.NewJsonResponseOrPanic(200, httpmock.File("body.json"))
+func NewJsonResponseOrPanic(status int, body any) *http.Response { // nolint: revive
```
Comment (maxatome) [inline response.go]: Unit test for this function is missing.

Comment (maxatome) [inline response.go]: New occurrences of nolint directive  should avoid space between `//` and `nolint`:
```suggestion
func NewJsonResponseOrPanic(status int, body any) *http.Response { //nolint: revive
```

**Review by maxatome — CHANGES_REQUESTED** [review]
Thanks for your PR. See my comments to go further.

Just as a note did you know that you can chain `Responder` using [`Then`](https://pkg.go.dev/github.com/jarcoal/httpmock#Responder.Then) method? It appeared after `ResponderFromMultipleResponses` implementation. Using it you can do:
```go
responder := httpmock.NewJsonResponderOrPanic(200, x).
  Then(httpmock.NewJsonResponderOrPanic(200, y)).
  Then(httpmock.NewJsonResponderOrPanic(200, z))
```

**Comment by maxatome** [comment]
Could you please also change the commit message to:
```
feat: add NewJsonResponseOrPanic function
```

**Comment by kilianpaquier** [comment]
> Thanks for your PR. See my comments to go further.
> 
> 
> 
> Just as a note did you know that you can chain `Responder` using [`Then`](https://pkg.go.dev/github.com/jarcoal/httpmock#Responder.Then) method? It appeared after `ResponderFromMultipleResponses` implementation. Using it you can do:
> 
> ```go
> 
> responder := httpmock.NewJsonResponderOrPanic(200, x).
> 
>   Then(httpmock.NewJsonResponderOrPanic(200, y)).
> 
>   Then(httpmock.NewJsonResponderOrPanic(200, z))
> 
> ```

Hi, yes I did see this builder function. We've been using it for the last two days, it's great too !

But I still think that `NewJsonResponseOrPanic` could be a useful addition, what do you think ?

**Comment by kilianpaquier** [comment]
> Could you please also change the commit message to:
> 
> ```
> 
> feat: add NewJsonResponseOrPanic function
> 
> ```

Will do !

File: response.go (hunk 4 above)
Comment (kilianpaquier) [reply response.go]: Shouldn't we apply this to the whole file for consistency ?

Comment (maxatome) [reply response.go]: I already did it 2 days ago.

File: response_test.go [hunk 5]
```diff
@@ -196,6 +196,41 @@ func TestNewJsonResponse(t *testing.T) {
...
+				assert.Cmp(response.Header.Get("Content-Type"), "application/json")
+				assertBody(assert, response, test.expected)
+			})
+		})
+	}
+
+	// Error case
+	assert.CmpPanic(
+		func() { _ = httpmock.NewJsonResponseOrPanic(200, func() {}) },
+		td.Ignore())
```
Comment (maxatome) [inline response_test.go]:
```suggestion
		td.NotNil())
```
or
```suggestion
		td.Contains("json: unsupported type"))
```
just to be sure we don't panic with a nil value.

File: response_test.go [hunk 6]
```diff
@@ -196,6 +196,41 @@ func TestNewJsonResponse(t *testing.T) {
...
+				assert.Cmp(response.StatusCode, 200)
+				assert.Cmp(response.Header.Get("Content-Type"), "application/json")
+				assertBody(assert, response, test.expected)
+			})
+		})
+	}
+
+	// Error case
+	assert.CmpPanic(
+		func() { _ = httpmock.NewJsonResponseOrPanic(200, func() {}) },
```
Comment (maxatome) [inline response_test.go]: We don't use blank identifier for one return value in this repo.
```suggestion
		func() { httpmock.NewJsonResponseOrPanic(200, func() {}) },
```

File: response.go [hunk 7]
```diff
@@ -586,6 +586,31 @@ func NewJsonResponse(status int, body any) (*http.Response, error) { // nolint:
 	return response, nil
 }
 
+// NewJsonResponseOrPanic is like [NewJsonResponse] but panics in case of error.
+//
+// It simplifies the call of [ResponderFromMultipleResponses], avoiding the
+// use of a temporary variable and an error check, and so can be used in such context:
```
Comment (maxatome) [inline response.go]: I didn't specified in the previous comment for the same line, but try to stay below 80 columns in comments:
```suggestion
// use of a temporary variable and an error check, and so can be used in such
// context:
```

Comment (kilianpaquier) [reply response.go]: Will keep that in mind !

**Review by maxatome — APPROVED** [review]

**Pull request merged**